pandas/core/arrays/dates.py

```diff
@@ -0,0 +1,191 @@
+        if isinstance(values, (ABCSeries, ABCIndexClass)):
+            values = values._values
+
+        if isinstance(values, type(self)):
+            values = values._data
+
+        if not isinstance(values, np.ndarray):
+            msg = (
+                f"Unexpected type '{type(values).__name__}'. 'values' must be "
+                "a DateArray ndarray, or Series or Index containing one of"
+                " those."
+            )
+            raise ValueError(msg)
+
+        if not self._is_compatible_dtype(values.dtype):
+            msg = (
+                f"The dtype of 'values' is incorrect. Must be one of {VALID_TYPES}."
+                f" Got {values.dtype} instead."
+            )
+            raise ValueError(msg)
+
+        if values.dtype == INTEGER_BACKEND:
+            values = values.view(D_DATETIME_DTYPE)
+        elif values.dtype != "datetime64[D]":
+            values = _to_date_values(values, copy)
+
+        if copy:
+            values = values.copy()
+
+        self._data = values
+
+    @staticmethod
+    def _is_compatible_dtype(dtype):
+        return (
+            is_integer_dtype(dtype)
+            or is_object_dtype(dtype)
+            or is_datetime64_dtype(dtype)
+            or dtype == "datetime64[D]"
+        )
+
+    @classmethod
+    def _simple_new(cls, values, **kwargs):
+        assert isinstance(values, np.ndarray)
+        if values.dtype == INTEGER_BACKEND:
+            values = values.view(D_DATETIME_DTYPE)
+
+        result = object.__new__(cls)
+        result._data = values
+        return result
+
+    @classmethod
+    def _from_sequence(cls, scalars, dtype=None, copy=False):
+        """
+        Construct a new ExtensionArray from a sequence of scalars.
+
+        Parameters
+        ----------
+        scalars : Sequence
+            Each element will be an instance of the scalar type for this
+            array, ``cls.dtype.type``.
+        dtype : dtype, optional
+            Construct for this particular dtype. This should be a Dtype
+            compatible with the ExtensionArray.
+        copy : bool, default False
+            If True, copy the underlying data.
+
+        Returns
+        -------
+        DateArray
+        """
+        if (
+            isinstance(scalars, np.ndarray)
+            and lib.infer_dtype(scalars, skipna=True) == "integer"
+        ):
+            values = scalars.astype(INTEGER_BACKEND)
+        elif is_integer_dtype(scalars):
+            values = scalars._data
+        else:
+            values = _to_date_values(scalars, copy)
+        return cls._simple_new(values)
+
+    @property
+    def dtype(self) -> ExtensionDtype:
+        return DateDtype()
+
+    def __iter__(self):
+        for date_data in self._data:
+            yield date_data
+
+    @property
+    def _box_func(self):
+        # TODO Implement Datestamp of a similar form in cython
+        return lambda x: Timestamp(x, freq="D", tz="utc")
```

Comment:
I _think_ we want to be timezone naive by default

Reply:
Yes that's why I have a todo to create a date stamp, but that will need to be implemented in cython, which I can do, I just wanted to get something working first.

Reply:
wouldnt we want datetime.date objects anyway (or Period[D] objects)

Reply:
@jbrockmendel Yeah probably this is just a place holder